New examples

tomcombriat opened the pull request:
Adding two new examples:
 - one on using envelope to control a filter (a few questions on that recently)
 - one to demonstrate how to make a simple RingModulator.

Still work in progress, the former does not sound synthy enough to my taste, and the later is very basic.

tomcombriat commented:
(Report jobs are failing because no previous versions of these examples were found to compare the memory, kinda ironic!)

tomcombriat commented:
Start to sound good! Here are some demo sounds:

## Filter_envelope:
https://github.com/user-attachments/assets/90759e9f-8ebc-43ac-8561-a1bba82b5115

## RingModulator
https://github.com/user-attachments/assets/530d9e54-2b3f-44af-bf25-02b0d9ebed22

The Filter envelope does not show in my list of Arduino examples, I do not understand why. The RM does work well, if someone has an idea...

tomcombriat commented:
Start to sound good! Here are some demo sounds:

## Filter_envelope:
https://github.com/user-attachments/assets/90759e9f-8ebc-43ac-8561-a1bba82b5115

## RingModulator
https://github.com/user-attachments/assets/530d9e54-2b3f-44af-bf25-02b0d9ebed22

The Filter envelope does not show in my list of Arduino examples, I do not understand why. The RM does work well, if someone has an idea...

sensorium commented:
Hi Tom, looking and sounding good.  Did you get the Filter showing up in examples yet?  I don't see any obvious clues...
Are you happy to merge, @tfry-git ?

sensorium commented:
It would be good to add them to the examples page on the site, too.

tomcombriat commented:
> It would be good to add them to the examples page on the site, too.

Yes indeed, I will try to do that!

tfry-git commented on examples/06.Synthesis/RingModulator/RingModulator.ino:
```diff
@@ -0,0 +1,85 @@
+    Mozzi is licensed under the GNU Lesser General Public Licence (LGPL) Version 2.1 or later.
+*/
+
+#define MOZZI_CONTROL_RATE 64  // Hz, powers of 2 are most reliable; 64 Hz is actually the default, but shown here, for clarity
```
Suggest to remove this line for simplicity.

tfry-git commented on examples/06.Synthesis/RingModulator/RingModulator.ino:
```diff
@@ -0,0 +1,85 @@
+uint8_t ring_mod_amount, smoothed_ring_mod_amount;
+
+UFix<8, 0> notes[4] = { 40 - 12, 52 - 12, 28 - 12, 30 - 12 };  // note played. Because of the ringModulation the oscillator is called *two times* 
+                                                              // hence produces a note which an octave to high, so we compensate for that here (12 midi notes makes an octave).
```
missing "is"

tfry-git commented on examples/07.Envelopes/Filter_Envelope/Filter_Envelope.ino:
```diff
@@ -0,0 +1,77 @@
+
+void setup() {
+  //Serial.begin(9600); // for Teensy 3.1, beware printout can cause glitches
+  Serial.begin(115200);
```
Suggest to remove Serial, here.

tfry-git commented on examples/07.Envelopes/Filter_Envelope/Filter_Envelope.ino:
```diff
@@ -0,0 +1,77 @@
+}
+
+
+unsigned int duration, attack, decay, sustain, release_ms;
```
Unused?

tomcombriat commented:
Good eye as always @tfry-git! Will correct soon, thanks!

tomcombriat commented:
Ok, I implemented the changes. I also renamed the RingModulator example, as, actually, this is not really a RM, but more a self modulator. The two examples shows in the IDE (cause was an hidden file locally).

@sensorium  For the website, I have to say I do not have a very clear view how all that is done… Are you using the scripts in extras/devscripts to automate the process of having all the examples copied to the gh-pages and generate the audio samples?

sensorium commented:
Hi Tom,

yes, I was using those scripts when Mozzi was developing, to check that all the examples worked locally after code changes and keep the pages in sync.  But with the checks you and Thomas have set up on Github and the more orderly development process, the scripts don't seem necessary anymore (I wonder whether to remove them to save confusion?).  I'd suggest to just edit the markdown for the webpage directly - what do you think?

tomcombriat commented:
Yes probably! 
I was just wondering about all the examples which are in the website (not the doxygen), that these pages seemed somehow to be generated automatically. Is this in the gh-page workflow or should I add them to this page manually?

sensorium commented:
I haven't made any Github workflows, if that's what you mean.
Any automation was local just with those scripts.  Don't go there!
You probably noticed the duplication of all the example sketches with their recordings.  I expect it's out of date and missing a few sketches - maybe I can regenerate it locally, if it looks like too much to update manually.  I'd have to set up a few things including hardware for recording, so I'm kind of hoping gradual changes by hand will be OK, but let me know.

tomcombriat commented:
Merging. Will add the needed doc soon in another PR!

tomcombriat commented:
Just to confirm, the changes made to `example-list.markdown` for example: https://github.com/sensorium/Mozzi/commit/d8e44a669cd4da136ddcd205d539f874e950ae3f were all done manually recently, or I guess the script `extras/devscripts/mozzi_generate_examples_page_buttons.sh` can be of good use? I'll try to use that one to regenerate the page, looks like there is nothing depending on the local architecture of the files in this one (which is not the case of `extras/devscripts/mozzi_generate_page.sh`)

tfry-git commented:
Yes, I have used an existing script (and probably it was extras/devscripts/mozzi_generate_examples_page_buttons.sh) for updating the examples page.

tomcombriat commented:
Thanks!
